examples/raytracing/CMakeLists.txt

```diff
@@ -1,3 +1,3 @@
 add_executable(ArborX_RayTracing.exe example_raytracing.cpp)
 target_link_libraries(ArborX_RayTracing.exe ArborX::ArborX Boost::program_options)
-add_test(NAME ArborX_RayTracing_Example COMMAND ./ArborX_RayTracing.exe --spheres=1000 --rays=50000 --L=200.0)
+add_test(NAME ArborX_RayTracing_Example COMMAND ./ArborX_RayTracing.exe --Nx=10 --Ny=10 --Nz=10 --Lx=1 --Ly=100000 --Lz=100000 --rays=1000)
```

Comment:
A more common GCD would be like 36x36x36, 48x48x48.
Do you set this aspect ratio of the domain size for any purpose?

Reply:
No, it's pretty random. Note that I use
```
./ArborX_RayTracing.exe --Nx=100 --Ny=100 --Nz=100 --rays=10
```
for all timings posted above.

Reply:
100x100x100 might be too large.
Maybe you can keep the number of rays to 1 million, but reduce the size to sth like 48x48x48.

Reply:
For 60x60x60, the profiling looks like
```
TOTAL TIME: 0.253154 seconds
TOP-DOWN TIME TREE:
<average time> <percent of total time> <percent time in Kokkos> <percent MPI imbalance> <remainder> <kernels per second> <number of calls> <name> [type]
===================
|-> 2.02e-01 sec 79.7% 37.0% 0.0% 60.3% 7.43e+01 1 second_approach [region]
|   |-> 6.93e-02 sec 27.4% 92.2% 0.0% 0.0% 1.73e+02 1 ArborX::query [region]
|   |   |-> 6.93e-02 sec 27.4% 92.2% 0.0% 0.7% 1.73e+02 1 ArborX::CrsGraphWrapper::query::spatial [region]
|   |       |-> 6.64e-02 sec 26.2% 96.0% 0.0% 1.0% 1.36e+02 1 ArborX::CrsGraphWrapper::two_pass [region]
|   |       |   |-> 4.11e-02 sec 16.2% 95.7% 0.0% 4.2% 7.30e+01 1 ArborX::CrsGraphWrapper::two_pass:second_pass [region]
|   |       |   |   |-> 3.93e-02 sec 15.5% 99.9% 0.0% 0.1% 2.54e+01 1 ArborX::BVH::query::spatial [region]
|   |       |   |   |   |-> 3.93e-02 sec 15.5% 100.0% 0.0% ------ 1 ArborX::TreeTraversal::spatial [for]
|   |       |   |-> 1.85e-02 sec 7.3% 99.8% 0.0% 0.1% 5.40e+01 1 ArborX::CrsGraphWrapper::two_pass::first_pass [region]
|   |       |   |   |-> 1.85e-02 sec 7.3% 99.9% 0.0% 0.1% 5.40e+01 1 ArborX::BVH::query::spatial [region]
|   |       |   |       |-> 1.85e-02 sec 7.3% 100.0% 0.0% ------ 1 ArborX::TreeTraversal::spatial [for]
|   |       |   |-> 6.01e-03 sec 2.4% 97.3% 0.0% 2.7% 6.65e+02 1 ArborX::CrsGraphWrapper::first_pass_postprocess [region]
|   |       |   |   |-> 5.75e-03 sec 2.3% 100.0% 0.0% ------ 1 ArborX::Algorithms::exclusive_scan [scan]
|   |       |-> 2.20e-03 sec 0.9% 7.2% 0.0% 92.8% 9.07e+02 1 ArborX::CrsGraphWrapper::query::spatial::compute_permutation [region]
|   |       |-> 2.70e-04 sec 0.1% 10.3% 0.0% 89.7% 3.70e+03 1 ArborX::CrsGraphWrapper::query::spatial::init_and_alloc [region]
|   |-> 9.95e-03 sec 3.9% 100.0% 0.0% ------ 1 deposit_energy [for]
|   |-> 8.45e-04 sec 0.3% 100.0% 0.0% ------ 1 ArborX::Algorithms::iota [for]
|-> 3.98e-02 sec 15.7% 94.0% 0.0% 0.0% 7.54e+01 1 first_approach [region]
|   |-> 3.98e-02 sec 15.7% 94.0% 0.0% 0.6% 7.55e+01 1 ArborX::BVH::query::nearest [region]
|       |-> 3.72e-02 sec 14.7% 100.0% 0.0% ------ 1 ArborX::Experimental::TreeTraversal::OrderedNearestPredicate [for]
|       |-> 2.31e-03 sec 0.9% 6.8% 0.0% 93.2% 8.66e+02 1 ArborX::BVH::query::nearest::compute_permutation [region]
|-> 5.69e-03 sec 2.2% 31.4% 0.0% 0.3% 1.06e+03 1 problem_setup [region]
|   |-> 5.33e-03 sec 2.1% 31.6% 0.0% 68.4% 9.39e+02 1 make_rays [region]
|   |   |-> 7.40e-04 sec 0.3% 100.0% 0.0% ------ 1 initialize_rays [for]
|   |   |-> 5.57e-04 sec 0.2% 100.0% 0.0% ------ 1 "Kokkos::Random_XorShift64::state"="Kokkos::Random_XorShift64::state_mirror" [copy]
|   |   |-> 3.37e-04 sec 0.1% 100.0% 0.0% ------ 1 "Kokkos::Random_XorShift64::locks"="Kokkos::Random_XorShift64::locks_mirror" [copy]
|   |-> 3.44e-04 sec 0.1% 29.2% 0.0% 70.8% 2.90e+03 1 make_grid [region]
|-> 2.08e-03 sec 0.8% 25.1% 0.0% 14.0% 3.36e+03 1 ArborX::BVH::BVH [region]
|   |-> 1.02e-03 sec 0.4% 2.4% 0.0% 97.6% 9.83e+02 1 ArborX::BVH::BVH::sort_linearized_order [region]
|   |-> 4.30e-04 sec 0.2% 89.3% 0.0% 17.7% 9.31e+03 1 ArborX::BVH::BVH::generate_hierarchy [region]
|   |   |-> 2.93e-04 sec 0.1% 100.0% 0.0% ------ 1 ArborX::TreeConstruction::generate_hierarchy [for]
```
which is pretty similar (3.98e-02s vs. 2.02e-01s).
For 48x48x48, we have
```
TOTAL TIME: 0.133111 seconds
TOP-DOWN TIME TREE:
<average time> <percent of total time> <percent time in Kokkos> <percent MPI imbalance> <remainder> <kernels per second> <number of calls> <name> [type]
===================
|-> 1.01e-01 sec 76.1% 35.3% 0.0% 60.4% 1.48e+02 1 second_approach [region]
|   |-> 3.60e-02 sec 27.1% 88.0% 0.0% 0.0% 3.33e+02 1 ArborX::query [region]
|   |   |-> 3.60e-02 sec 27.1% 88.0% 0.0% 1.2% 3.33e+02 1 ArborX::CrsGraphWrapper::query::spatial [region]
|   |       |-> 3.30e-02 sec 24.8% 95.9% 0.0% 1.2% 2.73e+02 1 ArborX::CrsGraphWrapper::two_pass [region]
|   |       |   |-> 1.70e-02 sec 12.8% 95.3% 0.0% 4.5% 1.77e+02 1 ArborX::CrsGraphWrapper::two_pass:second_pass [region]
|   |       |   |   |-> 1.62e-02 sec 12.1% 99.8% 0.0% 0.2% 6.19e+01 1 ArborX::BVH::query::spatial [region]
|   |       |   |   |   |-> 1.61e-02 sec 12.1% 100.0% 0.0% ------ 1 ArborX::TreeTraversal::spatial [for]
|   |       |   |-> 9.60e-03 sec 7.2% 99.6% 0.0% 0.2% 1.04e+02 1 ArborX::CrsGraphWrapper::two_pass::first_pass [region]
|   |       |   |   |-> 9.59e-03 sec 7.2% 99.7% 0.0% 0.3% 1.04e+02 1 ArborX::BVH::query::spatial [region]
|   |       |   |       |-> 9.56e-03 sec 7.2% 100.0% 0.0% ------ 1 ArborX::TreeTraversal::spatial [for]
|   |       |   |-> 5.94e-03 sec 4.5% 97.8% 0.0% 2.2% 6.73e+02 1 ArborX::CrsGraphWrapper::first_pass_postprocess [region]
|   |       |   |   |-> 5.72e-03 sec 4.3% 100.0% 0.0% ------ 1 ArborX::Algorithms::exclusive_scan [scan]
|   |       |-> 2.41e-03 sec 1.8% 4.5% 0.0% 95.5% 8.31e+02 1 ArborX::CrsGraphWrapper::query::spatial::compute_permutation [region]
|   |       |-> 2.43e-04 sec 0.2% 6.2% 0.0% 93.8% 4.11e+03 1 ArborX::CrsGraphWrapper::query::spatial::init_and_alloc [region]
|   |-> 3.70e-03 sec 2.8% 100.0% 0.0% ------ 1 deposit_energy [for]
|   |-> 3.96e-04 sec 0.3% 100.0% 0.0% ------ 1 ArborX::Algorithms::iota [for]
|-> 2.13e-02 sec 16.0% 91.1% 0.0% 0.0% 1.41e+02 1 first_approach [region]
|   |-> 2.13e-02 sec 16.0% 91.1% 0.0% 0.9% 1.41e+02 1 ArborX::BVH::query::nearest [region]
|       |-> 1.93e-02 sec 14.5% 100.0% 0.0% ------ 1 ArborX::Experimental::TreeTraversal::OrderedNearestPredicate [for]
|       |-> 1.80e-03 sec 1.4% 6.1% 0.0% 93.9% 1.11e+03 1 ArborX::BVH::query::nearest::compute_permutation [region]
|-> 5.38e-03 sec 4.0% 28.9% 0.0% 0.3% 1.12e+03 1 problem_setup [region]
|   |-> 4.81e-03 sec 3.6% 30.7% 0.0% 69.3% 1.04e+03 1 make_rays [region]
|   |   |-> 6.76e-04 sec 0.5% 100.0% 0.0% ------ 1 "Kokkos::Random_XorShift64::state"="Kokkos::Random_XorShift64::state_mirror" [copy]
|   |   |-> 4.09e-04 sec 0.3% 100.0% 0.0% ------ 1 initialize_rays [for]
|   |   |-> 3.39e-04 sec 0.3% 100.0% 0.0% ------ 1 "Kokkos::Random_XorShift64::locks"="Kokkos::Random_XorShift64::locks_mirror" [copy]
|   |-> 5.57e-04 sec 0.4% 13.9% 0.0% 86.1% 1.79e+03 1 make_grid [region]
|-> 1.77e-03 sec 1.3% 24.1% 0.0% 10.9% 3.96e+03 1 ArborX::BVH::BVH [region]
|   |-> 8.78e-04 sec 0.7% 2.4% 0.0% 97.6% 1.14e+03 1 ArborX::BVH::BVH::sort_linearized_order [region]
|   |-> 3.74e-04 sec 0.3% 82.9% 0.0% 24.9% 1.07e+04 1 ArborX::BVH::BVH::generate_hierarchy [region]
|   |   |-> 2.21e-04 sec 0.2% 100.0% 0.0% ------ 1 ArborX::TreeConstruction::generate_hierarchy [for]
|   |-> 2.44e-04 sec 0.2% 13.2% 0.0% 86.8% 4.11e+03 1 ArborX::BVH::BVH::compute_linear_ordering [region]
```
meaning 2.13e-02s vs. 1.01e-01s.